TSE's Review

carlos-ssh opened the pull request:
Hi i work on this project creating these: 

- Index, Edit, Create controller actions for the User model.
- Index, Edit, Create views created.
- Added the User model.

Thanks.

Carlos-SSH

Spayco commented on config/routes.rb:
```diff
@@ -1,3 +1,5 @@
-end
+  root 'users#new'
+
+  resources :users, only: %i[new create edit update index]
+end
```
* [ ] it should be `resources :users, only: [:new, :create, :edit, :update]` in config/routes.rb ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_route1.png) [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_route2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms)) no need for index on this project!

Spayco commented on app/views/users/_form.html.erb:
```diff
@@ -0,0 +1,23 @@
+<%= form_for @user do |f| %>
+  <% if @user.errors.any? %>
+    <% @user.errors.full_messages.each do |error| %>
+      <p><%= error %></p>
+    <% end %>
+  <% end %>
```
- [ ] no need for the if statement here, also please keep in mind that no logic should be in views, those go in helpers ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright)
* [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
* [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)

Spayco commented on app/models/user.rb:
```diff
@@ -0,0 +1,11 @@
+	# rubocop:disable Metrics/LineLength
+  EMAIL_REGEX = /(?:(?:\r\n)?[ \t])*(?:(?:(?:[^()<>@,;:\\".\[\] \000-\031]+(?:(?:(?:\r\n)?[ \t])+|\Z|(?=[\["()<>@,;:\\".\[\]]))|"(?:[^\"\r\\]|\\.|(?:(?:\r\n)?[ \t]))*"(?:(?:\r\n)?[ \t])*)(?:\.(?:(?:\r\n)?[ \t])*(?:[^()<>@,;:\\".\[\] \000-\031]+(?:(?:(?:\r\n)?[ \t])+|\Z|(?=[\["()<>@,;:\\".\[\]]))|"(?:[^\"\r\\]|\\.|(?:(?:\r\n)?[ \t]))*"(?:(?:\r\n)?[ \t])*))*@(?:(?:\r\n)?[ \t])*(?:[^()<>@,;:\\".\[\] ...
+  # rubocop:enable Metrics/LineLength
+  
+  class User < ApplicationRecord
+    attr_accessor :user
+
+    validates :username, presence: true
+    validates :email, presence: true, format: { with: EMAIL_REGEX, message: 'Not Valid' }
+    validates :password, presence: true
```
- [ ] `[OPTIONAL]` instead of writing that whole regex variable, you can use this:
```Ruby
validates :email, presence: true,  format: { with: URI::MailTo::EMAIL_REGEXP } 
```

Spayco reviewed (changes requested):
# Changes Requested 

Hello Team! 👋 
Awesome job so far!! 👍 you still need to make some changes though!

## Required Changes ♻️ 

- [ ] you need to have `<form>` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form1.png) [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms)) and `form_tag` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_tag2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms)) **commented out** in your  `_form.html.erb` or  `new.html.erb`,
- [ ] Update the title to describe what the branch is about ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_prs.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
- [ ] Enable your linters ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/))

### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

carlos-ssh commented:
**Hi @Spayco thanks for your answer and your feedback.
I do this changes to my project that you recommend me.**

- Added commented form tag out in _form.html.erb.
- Updated the title to describe the branch.
- Enabled linters in this project.
- Fixed issues related with my file config/routes.rb.
- Removed the if statement in  _form.html.erb.
- Removed my EMAIL_REGEX variable from my user model.

**_Greetings._**
**Carlos SSH**

voscarmv reviewed (approved):
# Project status: Approved ✨ 😎👍 💯 ✔️
### You may merge this branch and move forward ⛙🛣

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️
